src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs

```diff
@@ -389,6 +390,67 @@ private static ExtractMetadataInputModel ValidateInput(ExtractMetadataInputModel
             }
         }
 
+        private async Task<IEnumerable<IMethodSymbol>> GetAllExtensionMethodsAsync(IEnumerable<Project> projects)
+        {
+            List<INamespaceSymbol> namespaces = new List<INamespaceSymbol>();
+            foreach (var project in projects)
+            {
+                var compilation = await project.GetCompilationAsync();
+
+                if (compilation.Assembly.MightContainExtensionMethods)
+                {
+                    namespaces.AddRange(GetAllNamespaceMembers(compilation.GlobalNamespace));
+                }
+                var referencedAssemblies = from r in project.MetadataReferences
+                                           let a = (IAssemblySymbol)compilation.GetAssemblyOrModuleSymbol(r)
+                                           where a.MightContainExtensionMethods
+                                           select a;
+                var referencedNamespaces = (from s in referencedAssemblies
+                                            from n in GetAllNamespaceMembers(s.GlobalNamespace)
+                                            select n);
+                namespaces.AddRange(referencedNamespaces);
+            }
+            return from n in namespaces.Distinct()
```

Comment:
remove distinct()? I think same namespace in different assemblies should be different INamespaceSymbol.